cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -1827,4 +1874,68 @@ public PartialPath getStorageGroupPath(PartialPath path) throws StorageGroupNotS
       return super.getStorageGroupPath(path);
     }
   }
+
+  @Override
+  public void createAlignedTimeSeries(CreateAlignedTimeSeriesPlan plan) throws MetadataException {
+    PartialPath path = plan.getPrefixPath();
+    if (checkSgNodeAndPlanMajorVersion(path)) {
+      super.createAlignedTimeSeries(plan);
+    }
+  }
+
+  @Override
+  public void createTimeseries(CreateTimeSeriesPlan plan) throws MetadataException {
+    PartialPath path = plan.getPath();
+    if (checkSgNodeAndPlanMajorVersion(path)) {
+      super.createTimeseries(plan);
+    }
+  }
+
+  /**
+   * @param path the partial path
+   * @return true if the sg node major version == the path's major version; false if the sg node
+   *     major version > the path's major version; throw MetadataException if the sg node major
+   *     version < the path's major version.
+   * @throws MajorVersionNotEqualException sync meta leader failed and the
+   * @throws StorageGroupNotSetException some error occurred.
+   */
+  private boolean checkSgNodeAndPlanMajorVersion(PartialPath path)
+      throws MetadataException, StorageGroupNotSetException {
+    StorageGroupMNode node = null;
+    try {
+      node = getStorageGroupNodeByPath(path);
+    } catch (StorageGroupNotSetException e) {
+      syncMetaLeader();
+      node = getStorageGroupNodeByPath(path);
+    }
+
```

Comment:
so, why do not check whether node is null?

Reply:
The node can not be null, if the sg not found, it will throw `StorageGroupNotSetException` rather than null, so we do not need to check the node is null or not.